src/libstore/globals.hh

```diff
@@ -234,6 +234,13 @@ public:
           default is `true`.
         )"};
 
+    Setting<bool> fsyncStorePaths{this, false, "fsync-store-paths",
+        R"(
+        "Whether to call `fsync()` on store paths before registering them, to
+         flush them to disk. This improves robustness in case of system crashes,
+         but reduces performance. The default is `false`.
+        )"};
```

Comment:
`LocalStore` should use a local store option, not global option.

Reply:
Unfortunately, AFAIK, we don't have a convenient way to configure settings for the store. You don't want to set 
```
store = local?fsync-store-paths=1
```
in `nix.conf` because that will override the default value of `auto`, breaking the use of the daemon.

Also, overriding this setting when building via the daemon would be impossible.